src/raftstore/store/store.rs

```diff
@@ -2216,6 +2221,22 @@ impl<T: Transport, C: PdClient> Store<T, C> {
             error!("{} register compact cf-lock tick err: {:?}", self.tag, e);
         }
     }
+
+    fn register_clean_stale_peer_tick(&self, event_loop: &mut EventLoop<Self>) {
```

Comment:
why not move the timer to the region thread directly? Now we have already supporter timer for the worker.

Reply:
Because it will cause the channel `recv_timeout` error, and still don't know how to solve this bug.

Reply:
if we can't fix this bug, we can't use the recv_timeout. We should investigate it, not neglect it. 

Can we reproduce this with a simple program?

Reply:
No, for now we can't.
I am trying to compile a local `rustc` so that i can put some log in it to investigate this problem.

Reply:
so use it, reproduce the bug then fix the bug, can you show the error information?

Reply:
```
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:68
             at libstd/sys_common/backtrace.rs:57
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:380
   3: std::panicking::default_hook
             at libstd/panicking.rs:396
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:576
   5: std::panicking::begin_panic
             at libstd/panicking.rs:537
   6: std::panicking::begin_panic_fmt
             at libstd/panicking.rs:521
   7: <std::sync::mpsc::shared::Packet<T>>::decrement
             at /checkout/src/libstd/macros.rs:72
   8: <std::sync::mpsc::shared::Packet<T>>::recv
             at /checkout/src/libstd/sync/mpsc/shared.rs:232
   9: <std::sync::mpsc::Receiver<T>>::recv_deadline
             at /checkout/src/libstd/sync/mpsc/mod.rs:1387
  10: <std::sync::mpsc::Receiver<T>>::recv_timeout
             at /checkout/src/libstd/sync/mpsc/mod.rs:1300
  11: tikv::util::worker::fill_task_batch
             at ./src/util/worker/mod.rs:328
  12: tikv::util::worker::poll
             at ./src/util/worker/mod.rs:290
  13: <tikv::util::worker::Worker<T>>::start_with_timer::{{closure}}
             at ./src/util/worker/mod.rs:388
```

Reply:
any update for the panic? If not, we should still use the recv_timeout directly here.

Reply:
I am investigating using local `rustc`.